File: DataAlgo_forPy/Stacks_Queues/deque.js
class Deque {
  constructor() {
    this.items = []
  }

  isEmpty() {
    return this.items.length === 0
  }

  addFront(item) {
    this.items.push(item)
  }

  addRear(item) {
    this.items.unshift(item)
  }

  removeFront() {
    return this.items.pop()
  }

  removeRear() {
    return this.items.shift()
  }

  size() {
    return this.items.length
  }
}

const myDeque = new Deque()

myDeque.addFront("hello")
myDeque.addRear("world")

console.log(myDeque.size())

console.log(`${myDeque.removeFront()} ${myDeque.removeRear()}`)

console.log(myDeque.size())

File: DataAlgo_forPy/Stacks_Queues/deque.rb
class Deque
  def initialize
    @items = []
  end

  def is_empty?
    @items.empty?
  end

  def addFront(item)
    @items.unshift(item)
  end

  def addRear(item)
    @items.push(item)
  end

  def removeFront
    @items.unshift()
  end

  def removeRear
    @items.pop()
  end

  def size
    @items.length
  end
end

myDeque = Deque.new()

myDeque.addFront("hello")
myDeque.addRear("world")

p myDeque.size

p "#{myDeque.removeFront()} #{myDeque.removeRear()}"

p myDeque.size
